### PR TITLE
Add OpenTelemetry RabbitMQ support

### PR DESCRIPTION
**Pull request by jeffkreeftmeijer**
This patch adds support for `@opentelemetry/instrumentation-amqplib` by including it in the default instrumentation list. It comes with a rest case named `express-rabbitmq`, which is based on the already existing `express-redis`.

To do
===

- [x] Add instrumentation
- [x] Add integration test
- [x] Add to an appsignal/test-setups test setup (https://github.com/appsignal/test-setups/pull/202)
- [x] Add extractor in the agent (https://github.com/appsignal/appsignal-agent/pull/1132)
- [x] Write docs for it (https://github.com/appsignal/appsignal-docs/pull/1883)

### CODE REVIEW

**Comment by tombruijn** [comment]
Unassigning myself as a reviewer until I can test it in a test app.

File: src/client.ts [hunk 1]
```diff
@@ -45,6 +46,7 @@ import {
 import { SpanProcessor, TestModeSpanProcessor } from "./span_processor"
 
 const DefaultInstrumentations = {
+  "@opentelemetry/instrumentation-amqp": AmqplibInstrumentation,
```
Comment (unflxw) [inline src/client.ts:49]: Let's make this match the name of the actual instrumentation package.

```suggestion
  "@opentelemetry/instrumentation-amqplib": AmqplibInstrumentation,
```

File: test/express-rabbitmq/tests/spec/app_spec.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+RSpec.describe "Rabbitm app" do
```
Comment (unflxw) [inline test/express-rabbitmq/tests/spec/app_spec.rb:3]:
```suggestion
RSpec.describe "RabbitMQ app" do
```

(is this what it was supposed to say?)

**Review by unflxw — APPROVED** [review]
As mentioned before on conversations about this task, I really think we should amend the extractor to add `appsignal.ignore_trace_if_root` before releasing it. Otherwise these spans will become performance actions in uninstrumented contexts.

**Pull request merged**